akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -123,11 +123,21 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
     } else onBadProtocol
   }
 
+  /**
+   * @param ch connection header
+   * @param clh content-length
+   * @param cth content-type
+   * @param teh transfer-encoding
+   * @param e100c expect 100 continue
+   * @param hh host header seen
+   * @param resp parsing response headers
+   * @return
+   */
   @tailrec protected final def parseHeaderLines(input: ByteString, lineStart: Int, headers: ListBuffer[HttpHeader] = initialHeaderBuffer,
                                                 headerCount: Int = 0, ch: Option[Connection] = None,
                                                 clh: Option[`Content-Length`] = None, cth: Option[`Content-Type`] = None,
                                                 teh: Option[`Transfer-Encoding`] = None, e100c: Boolean = false,
-                                                hh: Boolean = false): StateResult =
+                                                hh: Boolean = false, resp: Boolean): StateResult =
```

Comment:
It would be simpler to just make that a `def isResponseParser` so it doesn't have to be passed around.